### PR TITLE
Oracle provider implementation

### LINKED ISSUE #2743: PushSecret for Oracle Vault (OCI) doesn't work
**Describe the bug**
I'm trying to set a PushSecret to push a specific secret to Oracle Vault, but it doesn't work.

**To Reproduce**
1. provide all relevant manifests

- Secret:
apiVersion: v1
data:
  teste: dGVzdGU=
  tls.crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUUvekNDQStlZ0F3SUJBZ0lTQTVWNngrcFlsVGdxU1B1T05GczNzalMxTUEwR0NTcUdTSWIzRFFFQkN3VUEKTURJeEN6QUpCZ05WQkFZVEFsVlRNUll3RkFZRFZRUUtFdzFNWlhRbmN5QkZibU55ZVhCME1Rc3dDUVlEVlFRRApFd0pTTXpBZUZ3MHlNekE1TWpjeE5ESXdNek5hRncweU16RXlNall4TkRJd016SmFNQ1V4SXpBaEJnTlZCQU1UCkduUmxjM1JsTWk1dlkya3VibUYwZFhKaFkyeHZkV1F1WTI5dE1JSUJJakFOQmdrcWhraUc5dzBCQVFFRkFBT0MKQVE4QU1JSUJDZ0tDQVFFQXVoaU9ldW81RWF1VXJWQjZXRHltbXkrVDdHRFlhZDdXWUU5ZW1vSUVVcWpqa2k3WAp1WXVMd2FiZitXNEJKdGY5cnhoS2NWaTk2VmRMdkFnQ3JXL2RaeVduaElydDB6ZTVwTXJobThkS2JCdEhmc3N4CnIvSHQxekxPSVNYMTl2MWVqbVBsTEw0STk5VU1uaXpxWHlydm1WZ0REUmxybS9yWmo0Q3BxZU1JWW40OVBJdU0KVjNwMFkwRk56UER1MWdBQ25jVU80NG9VQWVDcHVpVlBCTWZFNVo0ZURRK2EvSmppdGFYaW9SV1ZIVjkwUEJSQQo3Q0ExdG1sSFNST2JoZ3pkRmNFWUY2czBMdmZDeXRRcjdrTENiNjhVQmUwaVRFeWFhQVUwWGdRQy9VbktjRTR4Ci9KNFl6aW1PSjhBd0VZSXFMcXNxMXdaeE1TSndra05sekhwSkxRSURBUUFCbzRJQ0dqQ0NBaFl3RGdZRFZSMFAKQVFIL0JBUURBZ1dnTUIwR0ExVWRKUVFXTUJRR0NDc0dBUVVGQndNQkJnZ3JCZ0VGQlFjREFqQU1CZ05WSFJNQgpBZjhFQWpBQU1CMEdBMVVkRGdRV0JCUUpoeEZvQ0h6czB1R3piQnZiTWR4RUVKMHlBREFmQmdOVkhTTUVHREFXCmdCUVVMck1YdDFoV3k2NVFDVURtSDYrZGl4VEN4akJWQmdnckJnRUZCUWNCQVFSSk1FY3dJUVlJS3dZQkJRVUgKTUFHR0ZXaDBkSEE2THk5eU15NXZMbXhsYm1OeUxtOXlaekFpQmdnckJnRUZCUWN3QW9ZV2FIUjBjRG92TDNJegpMbWt1YkdWdVkzSXViM0puTHpBbEJnTlZIUkVFSGpBY2docDBaWE4wWlRJdWIyTnBMbTVoZEhWeVlXTnNiM1ZrCkxtTnZiVEFUQmdOVkhTQUVEREFLTUFnR0JtZUJEQUVDQVRDQ0FRSUdDaXNHQVFRQjFua0NCQUlFZ2ZNRWdmQUEKN2dCMUFMYysreVRmbkUyNmRmSTV4YnBZOUd4ZC9FTFBlcDgxeEo0ZENZRWw3YlNaQUFBQml0YzZQdGtBQUFRRApBRVl3UkFJZ01RV1dYTTgwK1lNYmtmL3ovV0ZtN3lueUUwL0sramFCOWRrdEk0Rk1ISzRDSUh6N2c0Rjh3K0pUCkQvS2pKSWZQU2x3bUhkMzJWNUNjM0ZlMnFPWll4WDZGQUhVQXJmZSsrbnovRU1pTG5UMmNIajRZYXJSbktWM1AKc1F3a3lvV0dOT3ZjZ29vQUFBR0sxem8vREFBQUJBTUFSakJFQWlBeUFBSTl5TjNMV0pIckw2dS9GUm54Zm4rRwpRdzZhSjh5eEVOYVNVZE9PZ1FJZ05zZkMwdml3dFlvdmJBZlRXM0xjVmtteGM2M21ISmIwbFlacWl3TEo5cHd3CkRRWUpLb1pJaHZjTkFRRUxCUUFEZ2dFQkFHTW5JazBQSFFhaStieEJ3eEY3KzQ2MllLbjBrMko1b1hYSVJDNmcKNy9GbVFwQ0M2K0Q4NkthQTN5cm5lNlNsR1B3U1p5MThWWlM0b2hOVGMwamJZWDU0Mm5iZUtzMW9ISUhSQjJYagpoQ3UrV0tCYXh6WXhwSFlpVURLY29oUjRuQXlmTnUrbkcvRDRlNmlZaDhYM21kcllEV2pSR0x0b3cyRm01aHF6CnJjeHMyM3Q4WEhOc1VqUXNmcWZpM2IrWjB1NjhjaklZR2tla0gxeXBWVWFoVzV5ZmZHSG9GRmttWHlNOGplOGkKOVR6QzhPbkxVdk9RQUxHbVN4YS9vUVpKcnFiSlVURm95cnQ3QjBJNkpaYXcyZVdCV3ZJaDRSMFAzaHlqOFVtMAo3bWwzWDdJVUVzU2FudUtEN01XZzhyaFhzb0w1bVo0eGRVdkloRE1iNUNmZGFvWT0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQotLS0tLUJFR0lOIENFUlRJRklDQVRFLS0tLS0KTUlJRkZqQ0NBdjZnQXdJQkFnSVJBSkVyQ0VyUERCaW5VL2JXTGlXblgxb3dEUVlKS29aSWh2Y05BUUVMQlFBdwpUekVMTUFrR0ExVUVCaE1DVlZNeEtUQW5CZ05WQkFvVElFbHVkR1Z5Ym1WMElGTmxZM1Z5YVhSNUlGSmxjMlZoCmNtTm9JRWR5YjNWd01SVXdFd1lEVlFRREV3eEpVMUpISUZKdmIzUWdXREV3SGhjTk1qQXdPVEEwTURBd01EQXcKV2hjTk1qVXdPVEUxTVRZd01EQXdXakF5TVFzd0NRWURWUVFHRXdKVlV6RVdNQlFHQTFVRUNoTU5UR1YwSjNNZwpSVzVqY25sd2RERUxNQWtHQTFVRUF4TUNVak13Z2dFaU1BMEdDU3FHU0liM0RRRUJBUVVBQTRJQkR3QXdnZ0VLCkFvSUJBUUM3QWhVb3pQYWdsTk1QRXV5TlZaTEQrSUx4bWFaNlFvaW5YU2FxdFN1NXhVeXhyNDVyK1hYSW85Y1AKUjVRVVZUVlhqSjZvb2prWjlZSThRcWxPYnZVN3d5N2JqY0N3WFBOWk9PZnR6Mm53V2dzYnZzQ1VKQ1dIK2pkeApzeFBuSEt6aG0rL2I1RHRGVWtXV3FjRlR6alRJVXU2MXJ1MlAzbUJ3NHFWVXE3WnREcGVsUURScks5TzhadXRtCk5IejZhNHVQVnltWitEQVhYYnB5Yi91QnhhM1NobGc5RjhmbkNidnhLL2VHM01IYWNWM1VSdVBNclNYQmlMeGcKWjNWbXMvRVk5NkpjNWxQL09vaTJSNlgvRXhqcW1BbDNQNTFUK2M4QjVmV21jQmNVcjJPay81bXprNTNjVTZjRwova2lGSGFGcHJpVjF1eFBNVWdQMTdWR2hpOXNWQWdNQkFBR2pnZ0VJTUlJQkJEQU9CZ05WSFE4QkFmOEVCQU1DCkFZWXdIUVlEVlIwbEJCWXdGQVlJS3dZQkJRVUhBd0lHQ0NzR0FRVUZCd01CTUJJR0ExVWRFd0VCL3dRSU1BWUIKQWY4Q0FRQXdIUVlEVlIwT0JCWUVGQlF1c3hlM1dGYkxybEFKUU9ZZnI1MkxGTUxHTUI4R0ExVWRJd1FZTUJhQQpGSG0wV2VaN3R1WGtBWE9BQ0lqSUdsajI2WnR1TURJR0NDc0dBUVVGQndFQkJDWXdKREFpQmdnckJnRUZCUWN3CkFvWVdhSFIwY0RvdkwzZ3hMbWt1YkdWdVkzSXViM0puTHpBbkJnTlZIUjhFSURBZU1CeWdHcUFZaGhab2RIUncKT2k4dmVERXVZeTVzWlc1amNpNXZjbWN2TUNJR0ExVWRJQVFiTUJrd0NBWUdaNEVNQVFJQk1BMEdDeXNHQVFRQgpndDhUQVFFQk1BMEdDU3FHU0liM0RRRUJDd1VBQTRJQ0FRQ0Z5azVIUHFQM2hVU0Z2TlZuZUxLWVk2MTFUUjZXClBUTmxjbFF0Z2FEcXcrMzRJTDlmekxkd0FMZHVPL1plbE43a0lKK203NHV5QStlaXRSWThrYzYwN1RrQzUzd2wKaWtmbVpXNC9SdlRaOE02VUsrNVV6aEs4akNkTHVNR1lMNkt2elhHUlNnaTN5TGdqZXdRdENQa0lWejZEMlFRegpDa2NoZUFtQ0o4TXF5SnU1emx6eVpNakF2bm5BVDQ1dFJBeGVrcnN1OTRzUTRlZ2RSQ25iV1NEdFk3a2grQkltCmxKTlhvQjFsQk1FS0lxNFFEVU9Yb1JnZmZ1RGdoamUxV3JHOU1MK0hiaXNxL3lGT0d3WEQ5UmlYOEY2c3c2VzQKYXZBdXZEc3p1ZTVMM3N6ODVLK0VDNFkvd0ZWRE52Wm80VFlYYW82WjBmK2xRS2MwdDhEUVl6azFPWFZ1OHJwMgp5Sk1DNmFsTGJCZk9EQUxadllIN243ZG8xQVpsczRJOWQxUDRqbmtEclFveEIzVXFROWhWbDNMRUtRNzN4RjFPCnlLNUdoRERYOG9WZkdLRjV1K2RlY0lzSDRZYVR3N21QM0dGeEpTcXYzKzBsVUZKb2k1TGM1ZGExNDlwOTBJZHMKaENFeHJvTDErN21yeUlrWFBlRk01VGdPOXIwcnZaYUJGT3ZWMnowZ3AzNVowK0w0V1BsYnVFak4vbHhQRmluKwpIbFVqcjhnUnNJM3FmSk9RRnkvOXJLSUpSMFkvOE9td3QvOG9UV2d5MW1kZUhtbWprN2oxbllzdkM5SlNRNlp2Ck1sZGxUVEtCM3poVGhWMStYV1lwNnJqZDVKVzF6YlZXRWtMTnhFN0dKVGhFVUczc3pnQlZHUDdwU1dUVVRzcVgKbkxSYndIT29xN2hId2c9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCi0tLS0tQkVHSU4gQ0VSVElGSUNBVEUtLS0tLQpNSUlGWURDQ0JFaWdBd0lCQWdJUVFBRjNJVGZVNlVLNDduYXFQR1FLdHpBTkJna3Foa2lHOXcwQkFRc0ZBREEvCk1TUXdJZ1lEVlFRS0V4dEVhV2RwZEdGc0lGTnBaMjVoZEhWeVpTQlVjblZ6ZENCRGJ5NHhGekFWQmdOVkJBTVQKRGtSVFZDQlNiMjkwSUVOQklGZ3pNQjRYRFRJeE1ERXlNREU1TVRRd00xb1hEVEkwTURrek1ERTRNVFF3TTFvdwpUekVMTUFrR0ExVUVCaE1DVlZNeEtUQW5CZ05WQkFvVElFbHVkR1Z5Ym1WMElGTmxZM1Z5YVhSNUlGSmxjMlZoCmNtTm9JRWR5YjNWd01SVXdFd1lEVlFRREV3eEpVMUpISUZKdmIzUWdXREV3Z2dJaU1BMEdDU3FHU0liM0RRRUIKQVFVQUE0SUNEd0F3Z2dJS0FvSUNBUUN0NkNSejlCUTM4NXVlSzFjb0hJZSszTGZmT0pDTWJqem1WNkI0OTNYQwpvdjcxYW03MkFFOG8yOTVvaG14RWs3YXhZLzBVRW11L0g5THFNWnNoZnRFelBMcEk5ZDE1MzdPNC94THhJWnBMCndZcUdjV2xLWm1ac2ozNDhjTCt0S1NJRzgrVEE1b0N1NGt1UHQ1bCtsQU9mMDBlWGZKbElJMVBvT0s1UENtK0QKTHRGSlY0eUFkTGJhTDlBNGpYc0RjQ0ViZGZJd1BQcVBydDNhWTZ2ckZrL0NqaEZMZnM4TDZQKzFkeTcwc250Swo0RXdTSlF4d2pRTXBvT0ZUSk93VDJlNFp2eEN6U293L2lhTmhVZDZzaHdlVTlHTng3QzdpYjF1WWdlR0pYRFI1CmJIYnZPNUJpZWViYnBKb3ZKc1hRRU9FTzN0a1FqaGI3dC9lbzk4ZmxBZ2VZanpZSWxlZmlONVlOTm5XZSt3NXkKc1IyYnZBUDVTUVhZZ2QwRnRDcldRZW1zQVhhVkNnL1kzOVc5RWg4MUx5Z1hiTktZd2FnSlpIZHVSemU2enF4WgpYbWlkZjNMV2ljVUdRU2srV1Q3ZEp2VWt5UkduV3FOTVFCOUdvWm0xcHpwUmJvWTdubjF5cHhJRmVGbnRQbEY0CkZRc0RqNDNRTHdXeVBudEtIRXR6QlJMOHh1cmdVQk44UTVOMHM4cDA1NDRmQVFqUU1OUmJjVGEwQjdyQk1EQmMKU0xlQ081aW1mV0NLb3FNcGdzeTZ2WU1FRzZLREEwR2gxZ1h4RzhLMjhLaDhoanRHcUVncWlOeDJtbmEvSDJxbApQUm1QNnpqelpON0lLdzBLS1AvMzIrSVZRdFFpMENkZDRYbitHT2R3aUsxTzV0bUxPc2JkSjFGdS83eGs5VE5EClR3SURBUUFCbzRJQlJqQ0NBVUl3RHdZRFZSMFRBUUgvQkFVd0F3RUIvekFPQmdOVkhROEJBZjhFQkFNQ0FRWXcKU3dZSUt3WUJCUVVIQVFFRVB6QTlNRHNHQ0NzR0FRVUZCekFDaGk5b2RIUndPaTh2WVhCd2N5NXBaR1Z1ZEhKMQpjM1F1WTI5dEwzSnZiM1J6TDJSemRISnZiM1JqWVhnekxuQTNZekFmQmdOVkhTTUVHREFXZ0JURXA3R2tleXh4Cit0dmhTNUIxLzhRVllJV0pFREJVQmdOVkhTQUVUVEJMTUFnR0JtZUJEQUVDQVRBL0Jnc3JCZ0VFQVlMZkV3RUIKQVRBd01DNEdDQ3NHQVFVRkJ3SUJGaUpvZEhSd09pOHZZM0J6TG5KdmIzUXRlREV1YkdWMGMyVnVZM0o1Y0hRdQpiM0puTUR3R0ExVWRId1ExTURNd01hQXZvQzJHSzJoMGRIQTZMeTlqY213dWFXUmxiblJ5ZFhOMExtTnZiUzlFClUxUlNUMDlVUTBGWU0wTlNUQzVqY213d0hRWURWUjBPQkJZRUZIbTBXZVo3dHVYa0FYT0FDSWpJR2xqMjZadHUKTUEwR0NTcUdTSWIzRFFFQkN3VUFBNElCQVFBS2N3QnNsbTcvRGxMUXJ0Mk01MW9HclMrbzQ0Ky95UW9ERlZEQwo1V3hDdTIrYjlMUlB3a1NJQ0hYTTZ3ZWJGR0p1ZU43c0o3bzVYUFdpb1c1V2xIQVFVN0c3NUsvUW9zTXJBZFNXCjlNVWdOVFA1MkdFMjRIR050TGkxcW9KRmxjRHlxU01vNTlhaHkyY0kycUJETEtvYmt4L0ozdldyYVYwVDlWdUcKV0NMS1loWGtjR2R0d2xmRlJqbEJ6NHBZZzFodG1mNVg2RFlPOEE0anF2MklsOURqWEE2VVNiVzFGelhTTHI5TwpoZThZNElXUzZ3WTdiQ2tqQ1dEY1JRSk1FaGc3NmZzTzN0eEUrRmlZcnVxOVJVV2hpRjFteXY0UTZXK0N5QkZDCkRmdnA3T09HQU42ZEVPTTQrcVI5c2Rqb1NZS0VCcHNyNkd0UEFRdzRkeTc1M2VjNQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==
  tls.key: LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQpNSUlFcGdJQkFBS0NBUUVBdWhpT2V1bzVFYXVVclZCNldEeW1teStUN0dEWWFkN1dZRTllbW9JRVVxampraTdYCnVZdUx3YWJmK1c0Qkp0ZjlyeGhLY1ZpOTZWZEx2QWdDclcvZFp5V25oSXJ0MHplNXBNcmhtOGRLYkJ0SGZzc3gKci9IdDF6TE9JU1gxOXYxZWptUGxMTDRJOTlVTW5penFYeXJ2bVZnRERSbHJtL3JaajRDcHFlTUlZbjQ5UEl1TQpWM3AwWTBGTnpQRHUxZ0FDbmNVTzQ0b1VBZUNwdWlWUEJNZkU1WjRlRFErYS9Kaml0YVhpb1JXVkhWOTBQQlJBCjdDQTF0bWxIU1JPYmhnemRGY0VZRjZzMEx2ZkN5dFFyN2tMQ2I2OFVCZTBpVEV5YWFBVTBYZ1FDL1VuS2NFNHgKL0o0WXppbU9KOEF3RVlJcUxxc3Exd1p4TVNKd2trTmx6SHBKTFFJREFRQUJBb0lCQVFDUnBqaVZrUTRiZTF3QgpEZFREakt5bEZ1YkNnY1cvanMzZ1VSNXM5VWVtM2xKR2pteis1bTM2b2IzNGZaVmZMbnQ1ZUJ5RGUvNGZTOWlWCmtQdHZmUGNmRE1TQW95K2lPYUlOalZCb1dPRXhuV1FtbzZmKzJlM1ptUVRsd3Mwd2czdG5IRjQ0MTM4ellVYmYKNGEySjBKUDRRMzhYMklaMWJTNnI5dnBUWWNLbE9teW51K0JBdGVmMGRRTzVVK2dpNkRYZTF0NGRDaWYzUmNCVwpaaW1YMm11L2NDRWhHVzhQYW1hVkhDbUVSME5UUVFjd2VUOGNxV2dTVE1IK0VDaWxzc1VsYTlsMlFVYUYzN2VDCkMxUU95alQ4TnFxVXd2QllLTk51dlFROUZUSDduQUw2elVuRHVtTHNSeEozR2FqYktTMWJXTU9VQW10T0NyR3AKWXMrZGFyK0JBb0dCQU9EMnBCaDFORDFwWnNpVnpHODZNeVFxUGdIblp6ajBhRnNVaWlqbXFMQUh3WGxqcHRpUQpwSkVNK0Fub3NQWmMzZnJHMitLV1JCa1lEcWNvQitaUlhGMUUybEtwTng0QStnRS9ZU0hRS3RUZE5rcUNvTjRwCmJ1alM0U0NyYktLT2ROcGkwd0tvQmVmeDZhL2Q1WlNpuu5nTzA1VVErNFB2VDk2NTQvUG1JSThoQW9HQkFOUEYKTE44a0lQMHhzb01iaGtzankrZ2t0NDJJS1JSUHZHcGJGc0pvb0VKRkZ3b1B1YjNYUExqMEp6enBQWXhPMGd1KwpvbmtNSGIwcDFpVUozMEFDYUtBb0JLcmhFUXd5Y1NPNCs0UWdGcWxVT2dSblhiVjlaQVdQN1pzaTJHSWU3OXNhCkJhYUo4Unh3YWxaanROTjFyV3VQV0EvalR6MjZpSVZNTDJNZ0lQU05Bb0dCQUkrekhOaityUkw1YkExL0dxcUcKQzdyelBNNnNpUlFwa3l0eThGbjJGUlQwcjFLaUpDYThKbEw5Yk1ldm1yS1VXYnRpSVgwWXJZRTJDODRjMVFiNwpPN3ZjU25WNDM5dExsbWNMbkRra0p1ZjRNa0RnTm5BVEdXU2hwK3JkZGJSVXVUSHk1SVVTM1VydkRXNG9tUWtiCmYrWSsya3FqSVhhQ2ZJdHlSTGs4YTluaEFvR0JBSlVMU0NxQWRndGRaK01vZ3hydFJiekw0cmZrNWV5YXRlUXcKMmRMWDZqclovMzRnVlR6TWVMcy9jMEpHY2dBSEZZUVlscWNWSlBVV3RkWDVuYnVpTmxySjc0Y0l3b1BRR0pybwp2WFJmVlRPMXlPT0NrSXlFaE5uNVUyU0ZZTCttUUcvM1g1UzNydyt4K3RWMGRUY0NzcHhxRmJPUTZ4dHA1bDNTCm9lTmxJR3lkQW9HQkFMT1dKeksxdVNuNmNrdzIvSW0zVlkyYnJwcStWdWVqVzVNZUtRb3JOeFp0QkJWSjZoTHMKSms1eTk3VHAvRGc5VnQyRGpMcHp4NVpUQTN3UDAwcytqZVltbHd2VFdHSHNBTWVlay9Kd216YWYrQ0IzeHlqRgpDazRjQVl0aUJidjNZR1BKNE9BM0srem05SjRobHpmMG5xWS9NU3YxR2FxZysyMkFFQUlha3ovdAotLS0tLUVORCBSU0EgUFJJVkFURSBLRVktLS0tLQo=
kind: Secret
metadata:
  name: wildcard-naturacloud-cert-secret-poc
  namespace: cert-manager
type: kubernetes.io/tls

- ClusterSecretStore:
apiVersion: external-secrets.io/v1beta1
kind: ClusterSecretStore
metadata:
  name: oke-sre-vault
spec:
  provider:
    oracle:
      region: sa-saopaulo-1
      vault: ocid1.vault.oc1.sa-saopaulo-1.ersdk666aahus.xxxxxxxxxxxxxxxyyyyyyyyyyyyyyyyyyyyyy
- PushSecret:
apiVersion: external-secrets.io/v1alpha1
kind: PushSecret
metadata:
  name: wildcard-naturacloud-cert-secret # Customisable
  namespace: cert-manager
spec:
  refreshInterval: 10s # Refresh interval for which push secret will reconcile
  secretStoreRefs: # A list of secret stores to push secrets to
    - name: oke-sre-vault
      kind: ClusterSecretStore
  selector:
    secret:
      name: wildcard-naturacloud-cert-secret-poc
  data:
    - match:
        secretKey: teste
        remoteRef:
          property: teste
          remoteKey: wildcard-naturacloud-cert-secret-poc
          
          
2. provide the Kubernetes and ESO version
K8s: 1.27
ESO: 0.9.5

**Expected behavior**
The expected behavior is the secret which is on K8S be pushed to Oracle Vault.

**Screenshots**
Logs:
Status:
  Conditions:
    Last Transition Time:  2023-09-28T19:29:32Z
    Message:               set secret failed: could not write remote ref teste to target secretstore oke-sre-vault: not implemented
    Reason:                Errored
    Status:                False
    Type:                  Ready
  Synced Push Secrets:
    ClusterSecretStore/oke-sre-vault:
Events:
  Type     Reason   Age                 From        Message
  ----     ------   ----                ----        -------
  Warning  Errored  11s (x14 over 53s)  pushsecret  set secret failed: could not write remote ref teste to target secretstore oke-sre-vault: not implemented

### PR DESCRIPTION
**Pull request by anders-swanson**
## Problem Statement

The Oracle Provider is missing the PushSecret, DeleteSecret, and GetAllSecrets function implementations. This pull request adds the missing implementations, so users of the Oracle Provider can get full functionality out of the external-secrets project.

## Related Issue

Fixes #2743 

## Proposed Changes

- Implement PushSecret method on the Oracle Provider, which creates or updates a secret in the specified Oracle Cloud Infrastructure (OCI) Vault.
- Implement DeleteSecret method on the Oracle Provider, which schedules secret deletion for an existing, non-deleting secret in the specified OCI Vault.
- Implement GetAllSecrets method on the Oracle Provider, which retrieves all secrets matching the remote ref in the specified OCI Vault.

## Checklist

- [ ] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [ ] All commits are signed with `git commit --signoff`
- [ ] My changes have reasonable test coverage
- [ ] All tests pass with `make test`
- [ ] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

**Comment by moolen** [comment]
I opened up a Oracle Cloud account to be able to do a proper review in the future. Is that pricing right or what is going on here? `$3.7k per hour` to launch a vault? HOLY GEEZ, no 🤯 



Going to take a look at the implementation without a testdrive :crossed_fingers:

**Comment by anders-swanson** [comment]
See https://www.oracle.com/cloud/price-list/#key-management for Oracle's public cloud price list, specifically key management. 

If you have any specific test cases you'd like me to run, I have access to a Oracle Cloud account and can perform all necessary tests.

**Review by moolen — COMMENTED** [review]
Did a testrun and it worked like a charm :tada: 
Left a comment regarding error sanitization, that was one thing that annoyed me and made it hard to spot the error.

File: pkg/provider/oracle/oracle.go [hunk 1]
```diff
@@ -77,19 +83,98 @@ type KmsVCInterface interface {
...
+	SecretAPIError
+)
+
+func (vms *VaultManagementService) PushSecret(ctx context.Context, value []byte, _ corev1.SecretType, _ *apiextensionsv1.JSON, remoteRef esv1beta1.PushRemoteRef) error {
+	secretName := remoteRef.GetRemoteKey()
+	encodedValue := base64.StdEncoding.EncodeToString(value)
+	sec, action, err := vms.getSecretBundleWithCode(ctx, secretName)
+	switch action {
+	case SecretNotFound:
+		_, err = vms.VaultClient.CreateSecret(ctx, vault.CreateSecretRequest{
```
[the remark this answers is not in the document]
Comment (anders-swanson) [reply pkg/provider/oracle/oracle.go:105]: How about this kind of error format?
> Vaults service failed to CreateSecret, HTTP status code 400: name already exists
> KmsVault service failed to GetVault, HTTP status code 400: Invalid OCID

[the remark this answers is not in the document]
Comment (anders-swanson) [reply pkg/provider/oracle/oracle.go:104]: Great, I added a commit which applies this formatting to SDK errors. Thanks for reviewing and testing.

**Review by moolen — APPROVED** [review]
LGTM! Errors look good now, that's a big UX improvement :tada:

**Pull request merged**